docs/design/2022-03-25-non-transactional-statement.md

```diff
@@ -0,0 +1,148 @@
+# Non-transactional statements
+
+- Author(s): [Ziqian Qin](http://github.com/ekexium)
+- Tracking issue: https://github.com/pingcap/tidb/issues/33485
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Test Design](#test-design)
+    * [Functional Tests](#functional-tests)
+    * [Scenario Tests](#scenario-tests)
+    * [Compatibility Tests](#compatibility-tests)
+    * [Benchmark Tests](#benchmark-tests)
+* [Impacts & Risks](#impacts--risks)
+* [Investigation & Alternatives](#investigation--alternatives)
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+A new syntax that transforms a given DML into a serial execution of mutually exclusive collectively exhaustive statements. 
+
+For example, `split on a limit 1000 delete from t` can be transformed into `delete from t where a between 1 and 1000` and `delete from t where a between 1001 and 2000`, assuming the min and max of `a` are respectively 1 and 2000.
+
+## Motivation or Background
+
+Users need to do a bulk delete or update using a single statement, while currently TiDB cannot satisfy the requirement because of the performance issue, transaction size limit, and compatibility issues with tools.
+
+A new syntax is proposed to work around the problem. A non-transactional DML contains a DML and information that are used to "split" the statement, thus it is equivalent to a sequence of DMLs which is not transactional since it does not provide atomicity and probably isolation.
+
+Different from the deprecated batch-DML, a non-transactional DML splits SQLs so every split SQL is a normal statement and does not risk data-index consistency.
+
+## Detailed Design
+
+### User Interface
+
+#### Syntax
+The syntax: `split on <column_name> limit <batch_size> <DML>`
+
+In the first step only `delete` is going to be supported, but `update` and `insert into select` are also worth considering.
+
+The split column must be indexed. There are no other constraints on the DML.
```

Comment:
You use the sql like 'select a from t where b < 1000 order by ISNULL(a,0,1), a', `ISNULL` in order by clause makes the index on column `a` invalid for order operation, Why we have this contraint?

Reply:
The split DML can make use of the index.